src/tiered/tiered_handle.c

```diff
@@ -8,6 +8,41 @@
 
 #include "wt_internal.h"
 
+/*
+ * __tiered_name_check --
+ *     Check if the given tiered table name exists in the shared storage. This is called from create
+ *     to check if this name had previously existed, was dropped and we're now trying to create the
+ *     same name. We cannot do that if shared objects exist for the old table.
+ */
+static int
+__tiered_name_check(WT_SESSION_IMPL *session, WT_TIERED *tiered)
+{
+    WT_DECL_RET;
+    WT_FILE_SYSTEM *bucket_fs;
+    u_int obj_count, i;
+    char **obj_files;
+    const char *name;
+
+    bucket_fs = tiered->bstorage->file_system;
+    name = tiered->iface.name;
+    WT_ASSERT(session, WT_PREFIX_MATCH(name, "tiered:"));
+    WT_PREFIX_SKIP(name, "tiered:");
+    /* See if this name exists in the shared storage. */
+    __wt_verbose(session, WT_VERB_TIERED, "NAME_CHECK: check for %s", name);
+    WT_ERR(bucket_fs->fs_directory_list(
```

Comment:
Do we have to prune the list that we get from `fs_directory_list`?  For example, if for table "abc" we check for objects with prefix "abc", and there exists "abcdef", wouldn't that match?

Reply:
Good catch. At some point I thought of that but then never went back to do that. I will fix it tomorrow.